OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/string_analysis/InterproceduralStringAnalysis.scala

```diff
@@ -0,0 +1,932 @@
+            val dependentVars = findDependentVars(state.computedLeanPath, stmts, uvar)
+            if (dependentVars.nonEmpty) {
+                dependentVars.keys.foreach { nextVar ⇒
+                    val toAnalyze = (nextVar, state.entity._2)
+                    dependentVars.foreach { case (k, v) ⇒ state.appendToVar2IndexMapping(k, v) }
+                    val ep = propertyStore(toAnalyze, StringConstancyProperty.key)
+                    ep match {
+                        case FinalP(p) ⇒ return processFinalP(state, ep.e, p)
+                        case _         ⇒ state.dependees = ep :: state.dependees
+                    }
+                }
+            } else {
+                attemptFinalResultComputation = true
+            }
+        } // If not a call to String{Builder, Buffer}.toString, then we deal with pure strings
+        else {
+            attemptFinalResultComputation = true
+        }
+
+        if (attemptFinalResultComputation) {
+            if (state.dependees.isEmpty && computeResultsForPath(state.computedLeanPath, state)) {
+                // Check whether we deal with the empty string; it requires special treatment as the
+                // PathTransformer#pathToStringTree would not handle it correctly (as
+                // PathTransformer#pathToStringTree is involved in a mutual recursion)
+                val isEmptyString = if (state.computedLeanPath.elements.length == 1) {
+                    state.computedLeanPath.elements.head match {
+                        case FlatPathElement(i) ⇒
+                            state.fpe2sci.contains(i) && state.fpe2sci(i).length == 1 &&
+                                state.fpe2sci(i).head == StringConstancyInformation.getNeutralElement
+                        case _ ⇒ false
+                    }
+                } else false
+
+                sci = if (isEmptyString) {
+                    StringConstancyInformation.getNeutralElement
+                } else {
+                    new PathTransformer(state.iHandler).pathToStringTree(
+                        state.computedLeanPath, state.fpe2sci
+                    ).reduce(true)
+                }
+            }
+        }
+
+        if (state.dependees.nonEmpty) {
+            getInterimResult(state)
+        } else {
+            InterproceduralStringAnalysis.unregisterParams(state.entity)
+            Result(state.entity, StringConstancyProperty(sci))
+        }
+    }
+
+    /**
+     * Continuation function for this analysis.
+     *
+     * @param state The current computation state. Within this continuation, dependees of the state
+     *              might be updated. Furthermore, methods processing this continuation might alter
+     *              the state.
+     * @return Returns a final result if (already) available. Otherwise, an intermediate result will
+     *         be returned.
+     */
+    private def continuation(
```

Comment:
please try to reduce the method size

Reply:
Done since dependency handling is revamped